programs/test/selftest.c

```diff
@@ -277,6 +278,14 @@ int main( int argc, char *argv[] )
     suites_tested++;
 #endif
 
+#if defined(MBEDTLS_CMAC_C) && ( defined(MBEDTLS_AES_C) || defined(MBEDTLS_DES_C) )
```

Comment:
I think this is redundant, you only need `defined(MBEDTLS_CMAC_C)` because if you do not have `MBEDTLS_AES_C` or `MBEDTLS_DES_C` if would not have passed compile time checks from `check_config.h`